### PR TITLE
CB-12145: Implement retire command

### PR DESCRIPTION
**Pull request by tiborpopovics**

### CODE REVIEW

**Review by doktoric — APPROVED** [review]

File: src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy [hunk 1]
```diff
@@ -218,6 +218,31 @@ trait ClusterService extends CommonService {
...
+   * @param host hostName where the component is installed to
+   * @param serviceName the service to retire
+   * @param componentName the name of the component
+   */
+  def int retire(List<String> hosts, String serviceName, String componentName)
+          throws URISyntaxException, ClientProtocolException, HttpResponseException, IOException {
+    def requestInfo = [
+            command   : 'RETIRE',
+            context   : "Execute retire command.",
+            parameters: ['excluded_hosts': hosts.join(',')]
```
Comment (horadla23) [inline src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy:232]: based on the example in the slack channel, i am not sure if this is needed

File: src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy [hunk 2]
```diff
@@ -218,6 +218,31 @@ trait ClusterService extends CommonService {
...
+    ]
+    def filter = [
+            ['service_name': serviceName, 'component_name': componentName]
```
Comment (horadla23) [inline src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy:235]: in the example, there is a third field, `hosts`
```
certs $ curl --insecure -u admin:admin1234 -X POST -H 'X-Requested-By:ambari' -H"Cookie: AMBARISESSIONID=node01wlt8eg7buea04wvjyq7yfs0812.node0" -d'{"RequestInfo":{"context":"Execute an action", "command" : "RETIRE"}, "Requests/resource_filters":[{"service_name": "NIFI", "component_name":"NIFI_MASTER","hosts":"ctr-e163-1618833847257-1055-01-000006.hwx.site"}] }' https://172.27.26.70/api/v1/clusters/mpatt2-1v78n8/requests
{
  "href" : "http://172.27.26.70/api/v1/clusters/mpatt2-1v78n8/requests/14",
  "Requests" : {
    "id" : 14,
    "status" : "Accepted"
  }
}
```

File: src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy (hunk 1 above)
Comment (tiborpopovics) [reply src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy]: fixed

File: src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy (hunk 2 above)
Comment (tiborpopovics) [reply src/main/groovy/com/sequenceiq/ambari/client/services/ClusterService.groovy]: yes, fixed

**Pull request merged**